src/analysis/retail/priest/holy/modules/talents/BottomRow/Lightweaver.tsx

```diff
@@ -30,44 +44,154 @@ class Lightweaver extends Analyzer {
       this.active = false;
       return;
     }
-    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.GREATER_HEAL), this.onHeal);
     this.addEventListener(
-      Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.LIGHTWEAVER_TALENT_BUFF),
-      this.onBuff,
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.GREATER_HEAL),
+      this.onHealCast,
     );
     this.addEventListener(
-      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.LIGHTWEAVER_TALENT_BUFF),
-      this.onBuff,
+      Events.cast.by(SELECTED_PLAYER).spell(SPELLS.FLASH_HEAL),
+      this.onFlashHealCast,
     );
   }
 
-  onHeal(Event: HealEvent) {
-    if (this.currentStacks > 0) {
-      this.healingDoneFromTalent += calculateEffectiveHealing(
-        Event,
-        HEALING_BONUS_PER_STACK * this.currentStacks,
-      );
-      this.overhealingDoneFromTalent += calculateOverhealing(
-        Event,
-        HEALING_BONUS_PER_STACK * this.currentStacks,
-      );
-
-      this.currentStacks = 0;
-      this.timesUsedBuffed += 1;
-    } else {
-      this.timesUsedUnbuffed += 1;
+  isHighOverheal(event: HealEvent) {
+    const rawHealing = (event.amount || 0) + (event.overheal || 0) + (event.absorbed || 0);
+    if (rawHealing === 0) {
+      return false;
+    }
+    return (event.overheal || 0) / rawHealing >= OVERHEAL_THRESHOLD;
+  }
+
+  onHealCast(event: CastEvent) {
+    // linked heal event exists
+    const healEvent = getHeal(event);
+    if (healEvent) {
+      this.totalHealCasts += 1;
+
+      if (isCastBuffedByLightweaver(event)) {
+        // calculate effective healing from bonus
+        this.healingDoneFromTalent += calculateEffectiveHealing(healEvent, HEALING_BONUS);
+        this.overhealingDoneFromTalent = calculateOverhealing(healEvent, HEALING_BONUS);
+      } else {
+        this.unbuffedHealCasts += 1;
+        // return early so we are not counting unbuffed heals for high overheal count
+        return;
+      }
+
+      if (this.isHighOverheal(healEvent)) {
+        this.highOverhealHealCasts += 1;
+      }
     }
   }
 
-  onBuff() {
-    this.totalStacks += 1;
-    if (this.currentStacks < MAX_STACKS) {
-      this.currentStacks += 1;
-    } else {
-      this.overcappedStacks += 1;
+  onFlashHealCast(event: CastEvent) {
+    // linked heal event exists
+    const healEvent = getHeal(event);
+    if (healEvent) {
+      this.totalFlashHealCasts += 1;
+      if (this.selectedCombatant.getBuffStacks(SPELLS.LIGHTWEAVER_TALENT_BUFF.id) < 2) {
+        if (this.isHighOverheal(healEvent)) {
+          this.highOverhealFlashHealCasts += 1;
+        }
```

Comment:
this seems like it should track high overheal casts regardless of whether it generates a stack, but i guess if you're only showing count you need to make sure you're not double-counting :thinking:

Reply:
i didn't do this because I'm considering any flash heal that doesn't generate a stack a bad cast(red bar), regardless of its overheal. the yellow bar is for high overheal casts that generate a stack if that makes sense